lang/src/org/partiql/lang/visitors/PartiQLVisitor.kt

```diff
@@ -235,6 +235,26 @@ internal class PartiQLVisitor(val ion: IonSystem, val customTypes: List<CustomTy
         insert(target, visit(ctx.value, PartiqlAst.Expr::class), conflictAction, metas)
     }
 
+    // TODO move from experimental; pending: https://github.com/partiql/partiql-docs/issues/27
+    override fun visitReplaceCommand(ctx: PartiQLParser.ReplaceCommandContext) = PartiqlAst.build {
+        insert(
+            target = visitSymbolPrimitive(ctx.symbolPrimitive()),
+            values = visit(ctx.value, PartiqlAst.Expr::class),
+            conflictAction = doReplace(excluded()),
+            metas = ctx.REPLACE().getSourceMetaContainer()
+        )
+    }
+
+    // TODO move from experimental; pending: https://github.com/partiql/partiql-docs/issues/27
+    override fun visitUpsertCommand(ctx: PartiQLParser.UpsertCommandContext) = PartiqlAst.build {
+        insert(
+            target = visitSymbolPrimitive(ctx.symbolPrimitive()),
+            values = visit(ctx.value, PartiqlAst.Expr::class),
+            conflictAction = doUpdate(excluded()),
```

Comment:
Assuming this is hardwired for now? Sorry I don't have more context on this.

Reply:
No worries—it's not, b/c `UPSERT INTO ...` is semantically equivalent to `INSERT INTO ... ON CONFLICT REPLACE EXCLUDED`.